### PR TITLE
1517 custom node images for runs tests

### PR DESCRIPTION
**Pull request by maryvictol**
This PR provides implementation for 1517_1, 1517_2 PipelineConfiguration tests

### CODE REVIEW

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/PipelineConfigurationNewTest.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import static com.epam.pipeline.autotests.ao.Primitive.STATUS;
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toSet;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotEquals;
+
+public class PipelineConfigurationNewTest extends AbstractSeveralPipelineRunningTest {
+    private final String pipeline1 = "pipe-config-" + Utils.randomSuffix();
+    private final String pipeline2 = "pipe-config-" + Utils.randomSuffix();
+    private String clusterNetworksConfig = "cluster.networks.config";
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/PipelineConfigurationNewTest.java]: The field is never used

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/PipelineConfigurationNewTest.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+import static com.epam.pipeline.autotests.ao.LogAO.taskWithName;
+import static com.epam.pipeline.autotests.ao.Primitive.EXEC_ENVIRONMENT;
+import static com.epam.pipeline.autotests.ao.Primitive.NODE_IMAGE;
+import static com.epam.pipeline.autotests.ao.Primitive.STATUS;
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toSet;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotEquals;
+
+public class PipelineConfigurationNewTest extends AbstractSeveralPipelineRunningTest {
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/PipelineConfigurationNewTest.java]: Let's rename this class to something CustomNodeImageTest

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/PipelineConfigurationNewTest.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    @TestCase("1517_1")
+    public void checkCustomNodeImageForThePipelineRun() {
+        final String[] cloudRegion = {""};
+        library()
+                .createPipeline(pipeline1)
+                .createPipeline(pipeline2)
+                .clickOnDraftVersion(pipeline1)
+                .configurationTab()
+                .editConfiguration("default", profile -> {
+                    cloudRegion[0] = profile
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/PipelineConfigurationNewTest.java]: cloudRegion is never used. Are we sure that need this?

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/PipelineConfigurationNewTest.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+                            .expandTab(EXEC_ENVIRONMENT)
+                            .getCloudRegion();});
+
+        String[] amis = navigationMenu()
+                .settings()
+                .switchToPreferences()
+                .getAmisFromClusterNetworksConfigPreference();
+        assertNotEquals(amiValue(amis[0]), testAmi = amiValue(amis[1]));
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/PipelineConfigurationNewTest.java]: Let's add a message to assert with ami values

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/PipelineConfigurationNewTest.java [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+        runsMenu()
+                .completedRuns()
+                .showLog(runID1517_1)
+                .logContainsMessage(logMess, format("Image: %s", testAmi))
+                .logContainsMessage(logMess, format("Specified in configuration image %s will be used", testAmi));
+    }
+
+    @Test (dependsOnMethods = {"checkCustomNodeImageForThePipelineRun"})
+    @TestCase("1517_2")
+    public void checkNodeReuseAfterTheCustomNodeImageRrun() {
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/PipelineConfigurationNewTest.java]: Rrun -> Run

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java [hunk 6]
```diff
@@ -52,6 +52,7 @@
 import static com.epam.pipeline.autotests.ao.Primitive.UPLOAD;
 import static com.epam.pipeline.autotests.utils.PipelineSelectors.button;
 import static com.epam.pipeline.autotests.utils.PipelineSelectors.buttonByIconClass;
+import static java.lang.String.format;
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java:55]: Update the license header, please

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java [hunk 7]
```diff
@@ -265,14 +266,17 @@ public FileEditingPopupAO editFile(final UnaryOperator<String> action) {
...
             Arrays.stream(expectedCode.split("\n"))
                     .map(String::trim)
                     .map(lineWithText)
                     .forEach(el -> el.should(exist));
             return this;
         }
 
+
+
+
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java]: Clean up, please

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/SettingsPageAO.java [hunk 8]
```diff
@@ -1026,6 +1027,22 @@ public PreferencesAO setSystemSshDefaultRootUserEnabled() {
             return this;
         }
 
+        public String[] getAmisFromClusterNetworksConfigPreference() {
+            searchPreference("cluster.networks.config");
+            String[] result = new String[2];
+            String[] strings = context().$(byClassName("CodeMirror-code"))
+                            .findAll(byClassName("CodeMirror-line")).texts().stream()
+                            .toArray(String[]::new);
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/SettingsPageAO.java]: Can be replaced with collection.toArray()

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/SettingsPageAO.java [hunk 9]
```diff
@@ -1026,6 +1027,22 @@ public PreferencesAO setSystemSshDefaultRootUserEnabled() {
...
+            IntStream.range(0, strings.length).forEach((i) -> {
+                if (strings[i].contains("\"instance_mask\":")) {
+                    if (strings[i].contains("\"*\"")) {
+                        result[0]=strings[i+1];
+                    } else {result[1]=strings[i+1];}
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/SettingsPageAO.java]: Add line breaks and spaces, please

**Review by kamyshova — CHANGES_REQUESTED** [review]

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/PipelineConfigurationNewTest.java (hunk 1 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/PipelineConfigurationNewTest.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/PipelineConfigurationNewTest.java (hunk 2 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/PipelineConfigurationNewTest.java]: Class has been renamed to CustomNodeImagesForRunsTest

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java (hunk 7 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/PipelineConfigurationNewTest.java (hunk 5 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/PipelineConfigurationNewTest.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java (hunk 6 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java:55]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/SettingsPageAO.java (hunk 8 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/SettingsPageAO.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/PipelineConfigurationNewTest.java (hunk 3 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/PipelineConfigurationNewTest.java]: cloudRegion has been added as parameter to getAmisFromClusterNetworksConfigPreference method

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/SettingsPageAO.java (hunk 9 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/SettingsPageAO.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/PipelineConfigurationNewTest.java (hunk 4 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/PipelineConfigurationNewTest.java]: Message has been added

**Review by kamyshova — APPROVED** [review]

**Pull request merged**